### PR TITLE
Scheduled Updates: Break out API endpoints into their own files

### PR DESCRIPTION
**Pull request by obenland**
## Proposed changes:

* Creates new files for each API endpoint path.
* Moves capability functionality into endpoint.
* Formalizes package dependencies.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

This will affect every aspect of the Scheduled Updates functionality.

* Navigate to the Scheduled Updates screen in Calypso and make sure the cap check endpoint works.
* Create a schedule.
* Downgrade at least one plugin.
* Run that schedule and make sure it succeeds.
* Check that the logs look right.
* Update a schedule.
* Make sure the logs get migrated.
* Delete one of the plugins. 
* Make sure the schedule gets updated.
* Delete a schedule.
* Make sure the logs are deleted as well.

### CODE REVIEW

**Comment by obenland** [comment]
@anomiex Is there anything I'm missing that might cause the phan test to show errors for jetpack-mu-wpcom?

**Comment by anomiex** [comment]
You're adding some missing package dependencies to `projects/packages/scheduled-updates/composer.json`, which is a good thing. That causes Composer to also include those packages in `packages/jetpack-mu-wpcom` since that depends on scheduled-updates, which in turn fixes a bunch of PhanUndeclaredClass errors but also means that the mis-typing of `Automattic\Jetpack\Connection\Client::wpcom_json_api_request_as_blog()`'s `$body` parameter is now detected.

We've got #37035 in progress for fixing missing deps in `packages/jetpack-mu-wpcom` directly and #37056 for fixing `wpcom_json_api_request_as_blog()`, if you want to wait for those.

**Comment by obenland** [comment]
Ah, perfect, thank you!

**Review by ouikhuan — DISMISSED** [review]
Tested with all the scenarios and works as expected, great work 🎉

File: projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-capabilities.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+	/**
+	 * The base of this controller's route.
+	 *
+	 * @var string
+	 */
+	public $rest_base = 'update-schedules';
+
+	/**
+	 * WPCOM_REST_API_V2_Endpoint_Atomic_Hosting_Update_Schedule constructor.
```
Comment (zaerl) [inline projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-capabilities.php]: Typo: `WPCOM_REST_API_V2_Endpoint_Update_Schedules_Capabilities`

File: projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-logs.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+	/**
+	 * The base of this controller's route.
+	 *
+	 * @var string
+	 */
+	public $rest_base = 'update-schedules';
+
+	/**
+	 * WPCOM_REST_API_V2_Endpoint_Atomic_Hosting_Update_Schedule constructor.
```
Comment (zaerl) [inline projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-logs.php]: Typo: `WPCOM_REST_API_V2_Endpoint_Update_Schedules_Logs`

File: projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-status.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+	/**
+	 * The base of this controller's route.
+	 *
+	 * @var string
+	 */
+	public $rest_base = 'update-schedules';
+
+	/**
+	 * WPCOM_REST_API_V2_Endpoint_Atomic_Hosting_Update_Schedule constructor.
```
Comment (zaerl) [inline projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-status.php]: Typo: `WPCOM_REST_API_V2_Endpoint_Update_Schedules_Status`

File: projects/packages/scheduled-updates/tests/php/bootstrap.php [hunk 4]
```diff
@@ -15,3 +15,8 @@
  * Load WorDBless.
  */
 \WorDBless\Load::load();
+
+/**
+ * Load rest api endpoints.
```
Comment (zaerl) [inline projects/packages/scheduled-updates/tests/php/bootstrap.php]: Let's use `Load REST API endpoints.`

File: projects/packages/scheduled-updates/tests/php/class-wpcom-rest-api-v2-endpoint-update-schedules-capabilities-test.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+/**
+ * Test class for WPCOM_REST_API_V2_Endpoint_Update_Schedules_Capabilities.
+ *
+ * @package automattic/scheduled-updates
+ */
+
+use Automattic\Jetpack\Scheduled_Updates;
+
+/**
+ * Test class for WPCOM_REST_API_V2_Endpoint_Update_Schedules.
```
Comment (zaerl) [inline projects/packages/scheduled-updates/tests/php/class-wpcom-rest-api-v2-endpoint-update-schedules-capabilities-test.php]: Typo: `WPCOM_REST_API_V2_Endpoint_Update_Schedules_Capabilities`

**Review by zaerl — APPROVED** [review]
Great work here! The codebase grew quickly. Did was needed. Performed all the tests and everything is where it should be. 🚢

**Pull request merged**